server/service/apple_mdm.go

```diff
@@ -2985,7 +2993,24 @@ func (svc *MDMAppleCheckinAndCommandService) GetToken(_ *mdm.Request, _ *mdm.Get
 //
 // [1]: https://developer.apple.com/documentation/devicemanagement/commands_and_queries
 func (svc *MDMAppleCheckinAndCommandService) CommandAndReportResults(r *mdm.Request, cmdResult *mdm.CommandResults) (*mdm.Command, error) {
+	fmt.Println(">>>>> COMMANDANDREPORTRESULTS!", r.ID, cmdResult.CommandUUID, cmdResult.Status, r.Params)
+
 	if cmdResult.Status == "Idle" {
+		// NOTE: iPhone/iPad devices that are still enroled in Fleet's MDM but have
+		// beeen deleted from Fleet (no host entry) will still send checkin
+		// requests from time to time. Those should be Idle requests without a
+		// CommandUUID. As stated in tickets #22941 and #22391, Fleet iDevices
+		// should be re-created when they checkin with MDM.
+
+		// TODO(iosrevive): Step 2: identify via the new nano_devices.platform if
+		// this is an iDevice that doesn't have a corresponding host entry, and if
+		// so, trigger a refetch device info for it (we can't create the host entry
+		// here as we don't have any other device information besides the UDID).
+		// I.e. the equivalent of:
+		// https://github.com/fleetdm/fleet/blob/769fd37aafe23e0c04e014bbe1a34804b8f1d348/server/service/hosts.go#L1111-L1114
+		// We can't insert the requested command in host_mdm_commands because the
+		// host entry doesn't exist yet.
```

Comment:
Note to self, seems like we could simplify this a bit with the `nano_devices.authenticate` XML field, we already have the info needed to create the host (same info as we have to trigger the MDM Reset lifecycle stage). We have the serial, the uuid, the model comes from the ProductName XML field, and the platform we will now store (but could also deduce from the ProductName).